gnoi/gnoi_test.go

```diff
@@ -0,0 +1,197 @@
+		prevTime, err := ygnmi.Get(context.Background(), c, ocpath.Root().System().BootTime().State())
+		if err != nil {
+			t.Fatal(err)
+		}
+		if _, err := s.Reboot(ctx, &spb.RebootRequest{}); err != nil {
+			t.Fatal(err)
+		}
+		afterTime, err := ygnmi.Get(context.Background(), c, ocpath.Root().System().BootTime().State())
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !(prevTime < afterTime) {
+			t.Errorf("boot time did not update after reboot")
+		}
+	})
+
+	t.Run("one-second-delay", func(t *testing.T) {
+		prevTime, err := ygnmi.Get(context.Background(), c, ocpath.Root().System().BootTime().State())
+		if err != nil {
+			t.Fatal(err)
+		}
+		const delay = 1e9
+		if _, err := s.Reboot(ctx, &spb.RebootRequest{Delay: delay}); err != nil {
+			t.Fatal(err)
+		}
+
+		var afterTime uint64
+		tryN := 50
+		for i := 0; i != tryN; i++ {
+			var err error
+			afterTime, err = ygnmi.Get(context.Background(), c, ocpath.Root().System().BootTime().State())
+			if err != nil {
+				t.Fatal(err)
+			}
+			if prevTime < afterTime {
+				break
+			}
+			time.Sleep(100 * time.Millisecond)
+		}
+		if !(prevTime < afterTime) {
+			t.Errorf("boot time did not update after reboot")
+		}
+	})
+
+	t.Run("cancel-no-pending", func(t *testing.T) {
+		if _, err := s.CancelReboot(ctx, &spb.CancelRebootRequest{}); err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	t.Run("cancel-pending", func(t *testing.T) {
+		prevTime, err := ygnmi.Get(context.Background(), c, ocpath.Root().System().BootTime().State())
+		if err != nil {
+			t.Fatal(err)
+		}
+		const delay = 120e9
+		if _, err := s.Reboot(ctx, &spb.RebootRequest{Delay: delay}); err != nil {
+			t.Fatal(err)
+		}
+		if _, err := s.CancelReboot(ctx, &spb.CancelRebootRequest{}); err != nil {
+			t.Fatal(err)
+		}
+
+		afterTime, err := ygnmi.Get(context.Background(), c, ocpath.Root().System().BootTime().State())
+		if err != nil {
+			t.Fatal(err)
+		}
+		if prevTime != afterTime {
+			t.Errorf("boot did not get cancelled")
+		}
+
+		// Couple more no-op cancels.
+		if _, err := s.CancelReboot(ctx, &spb.CancelRebootRequest{}); err != nil {
+			t.Fatal(err)
+		}
+		if _, err := s.CancelReboot(ctx, &spb.CancelRebootRequest{}); err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	t.Run("reboot-while-pending", func(t *testing.T) {
+		const delay = 120e9
+		if _, err := s.Reboot(ctx, &spb.RebootRequest{Delay: delay}); err != nil {
+			t.Fatal(err)
+		}
+		if _, err := s.Reboot(ctx, &spb.RebootRequest{}); status.Convert(err).Code() != codes.AlreadyExists {
+			t.Fatalf("Expected AlreadyExists error, got %v", err)
+		}
+		if _, err := s.Reboot(ctx, &spb.RebootRequest{Delay: 1}); status.Convert(err).Code() != codes.AlreadyExists {
+			t.Fatalf("Expected AlreadyExists error, got %v", err)
+		}
+		if _, err := s.CancelReboot(ctx, &spb.CancelRebootRequest{}); err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	closeProximityTests := []struct {
+		desc  string
+		delay uint64
+	}{{
+		desc:  "cancel-while-possibly-pending-1ns",
+		delay: 1e0,
```

Comment:
nit: use time.Duration in the test case and delay.Unix().Nanos() in the req

Reply:
Done.